torch/distributed/fsdp/_init_utils.py

```diff
@@ -395,9 +395,7 @@ def _init_device_handle(
                         f"FSDP does not support modules with different device types "
                         f"but got params on {determined_device.type} and {param.device.type}"
                     )
-        determined_device = determined_device or torch.device(
-            "cuda", torch.cuda.current_device()
-        )
+        determined_device = determined_device or torch._C._get_accelerator()
```

Comment:
I agree with @wconstab.
https://github.com/pytorch/pytorch/blob/faabda4fc9e3c033aca7b7aa921fba21ebae06d5/torch/csrc/Module.cpp#L2030
https://github.com/pytorch/pytorch/blob/faabda4fc9e3c033aca7b7aa921fba21ebae06d5/torch/__init__.py#L2074

Should we be using public APIs only here like `get_device_module`? cc: @albanD

Reply:
Sorry I missed that notification.

There were some debates in the original design about exposing this to the end users on the python side. That's why it doesn't exist as of today.

The idea is we really want this device to be used only as the default value for different APIs to make their default better. We don't want users setting this as the default device in general.

For this particular change, I think it's ok to use a private API and document it like we document pin_memory, by pointing back to https://pytorch.org/docs/main/torch.html#accelerators.
Also this API will return CPU when no accelerator is available. So it might not actually have the behavior you're looking for? We can definitely refine the API if needed.

Reply:
@albanD Thanks for your reply. Yes, it is not the expected behavior if this API return CPU when no accelerator is available. I think we can just pass `check=True` and let it raise error when no accelerator is available. I have modifed my code, could you review it again? @albanD @awgu